tests/e2e/happy_path_test.go

```diff
@@ -85,18 +87,29 @@ var _ = Describe("MCP Gateway Happy Path", func() {
 
 		By("Setting up port-forward to broker for status check")
 		statusPortForwardCmd := setupPortForward("mcp-broker-router", SystemNamespace, "18081:8080")
-		defer statusPortForwardCmd.Process.Kill()
+		defer func() {
```

Comment:
lots of additional debugging and error catching in this file while I was struggling with some failures related to my environment.